src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyExecutionManagerWithDataCollection.cs

```diff
@@ -84,22 +72,11 @@ public override void Initialize()
                     this.dataCollectionPort = dataCollectionParameters.DataCollectionEventsPort;
                 }
             }
-            catch (Exception ex)
+            catch (Exception)
             {
-                try
-                {
-                    // On failure in calling BeforeTestRunStart, call AfterTestRunEnd to end DataCollectionProcess
-                    this.ProxyDataCollectionManager.AfterTestRunEnd(isCanceled: true, runEventsHandler: this.DataCollectionRunEventsHandler);
-                    throw;
-                }
-                finally
-                {
-                    // There is an issue with Data Collector, skipping data collection and continuing with test run.
-                    if (EqtTrace.IsErrorEnabled)
-                    {
-                        EqtTrace.Error("ProxyExecutionManagerWithDataCollection: Error occured while communicating with DataCollection Process: {0}", ex.Message);
-                    }
-                }
+                // On failure in calling BeforeTestRunStart, call AfterTestRunEnd to end DataCollectionProcess
```

Comment:
This is conditional, if there is any exception, then only we will call the AfterTestRunComplete with cancel as true, otherwise not.